server/controllers/convo.js

```diff
@@ -0,0 +1,172 @@
+const asyncHandler = require("express-async-handler");
+const mongoose = require('mongoose');
+const Message = require('../models/Message');
+const Conversation = require('../models/Convo');
+
+module.exports = {
+  getAllConvos: asyncHandler(async (req, res) => {
+    let from = mongoose.Types.ObjectId(req.user.id);
+    Conversation.aggregate([{
+        $lookup: {
+          from: 'user',
+          localField: 'recipients',
+          foreignField: '_id',
+          as: 'recipientObj',
+        },
+      },
+     ])
+      .match({
+        recipients: {
+          $all: [{
+            $elemMatch: {
+              $eq: from
+            }
+          }]
+        }
+      })
+      .project({
+        'recipientObj.password': 0,
+        'recipientObj.__v': 0,
+      })
+      .exec(async (err, conversations) => {
+        if (err) {
+          console.log(err);
+          res.setHeader('Content-Type', 'application/json');
+          res.end(JSON.stringify({
+            message: 'Failure'
+          }));
+          res.sendStatus(500);
+        } else {
+          const populated = await Conversation.populate(conversations, {path: "recipients", select: '-__v -password -register_date'})
+          res.send(populated);
+        }
+      });
+  }),
+  getOneConvo: asyncHandler(async (req, res) => {
+    let user1 = mongoose.Types.ObjectId("60efa11241aa990dec93324e");
```

Comment:
Please use `req.body` to get user id's here

Reply:
I will fix this. I hard coded id's in for testing purposes.

Reply:
This is fixed